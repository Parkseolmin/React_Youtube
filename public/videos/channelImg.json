{
   "kind": "youtube#channelListResponse",
   "etag": "eiqKSaQ_2IqEwQBhzhQ3Aw-JuoQ",
   "pageInfo": {
      "totalResults": 1,
      "resultsPerPage": 5
   },
   "items": [
      {
         "kind": "youtube#channel",
         "etag": "xsJLLnQ06iJXV_rfQsPtHjoudiM",
         "id": "UCASXEOh4LJxWdk2hZnsfNYw",
         "snippet": {
            "title": "고요비koyobi",
            "description": "고요베이크샵을 운영하고 있습니다\n@koyobimood\n@koyobake\n💌Koyobi@sandbox.co.kr",
            "customUrl": "@koyobi",
            "publishedAt": "2013-11-09T14:02:11Z",
            "thumbnails": {
               "default": {
                  "url": "https://yt3.ggpht.com/ytc/AIdro_kMWUTfoQXAyIjWM2pDk0KRoGXXMjaZF1rKjtvX=s88-c-k-c0x00ffffff-no-rj",
                  "width": 88,
                  "height": 88
               },
               "medium": {
                  "url": "https://yt3.ggpht.com/ytc/AIdro_kMWUTfoQXAyIjWM2pDk0KRoGXXMjaZF1rKjtvX=s240-c-k-c0x00ffffff-no-rj",
                  "width": 240,
                  "height": 240
               },
               "high": {
                  "url": "https://yt3.ggpht.com/ytc/AIdro_kMWUTfoQXAyIjWM2pDk0KRoGXXMjaZF1rKjtvX=s800-c-k-c0x00ffffff-no-rj",
                  "width": 800,
                  "height": 800
               }
            },
            "localized": {
               "title": "고요비koyobi",
               "description": "고요베이크샵을 운영하고 있습니다\n@koyobimood\n@koyobake\n💌Koyobi@sandbox.co.kr"
            },
            "country": "KR"
         },
         "statistics": {
            "viewCount": "21298702",
            "subscriberCount": "157000",
            "hiddenSubscriberCount": false,
            "videoCount": "106"
         },
         "brandingSettings": {
            "channel": {
               "title": "고요비koyobi",
               "description": "고요베이크샵을 운영하고 있습니다\n@koyobimood\n@koyobake\n💌Koyobi@sandbox.co.kr",
               "country": "KR"
            },
            "image": {
               "bannerExternalUrl": "https://yt3.googleusercontent.com/Q7sm1fP506JLWqV2vowIU652B-Hc0t7mEvzHUC_En9yRbsN4frIhbm8Fxpd0QqRJmfvnv6sW"
            }
         }
      }
   ]
}
