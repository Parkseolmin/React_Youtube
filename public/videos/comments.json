{
   "kind": "youtube#commentThreadListResponse",
   "etag": "_POhe4y79w0hoMb4czP6w1h7gJM",
   "nextPageToken": "Z2V0X25ld2VzdF9maXJzdC0tQ2dnSWdBUVZGN2ZST0JJRkNJY2dHQUFTQlFpb0lCZ0FFZ1VJaVNBWUFCSUZDSjBnR0FFU0JRaUlJQmdBR0FBaURRb0xDTGZVMWE0R0VLaVBxZ1k=",
   "pageInfo": {
      "totalResults": 10,
      "resultsPerPage": 10
   },
   "items": [
      {
         "kind": "youtube#commentThread",
         "etag": "ex28uFYeENNY7-OjQXmwKgVmVKE",
         "id": "UgwOEtuZtUMtdJgAs8Z4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "m4l0hfyF0-li5ot6OF6evcRPYSI",
               "id": "UgwOEtuZtUMtdJgAs8Z4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "너무 뿌옇게나오는느낌",
                  "textOriginal": "너무 뿌옇게나오는느낌",
                  "authorDisplayName": "@uuooo0920",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/ytc/AIdro_n7Ywg_fIz9yt7STnNv8-qEyDmhbGc8lBEYKmystg=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@uuooo0920",
                  "authorChannelId": {
                     "value": "UCyu4c9V9TelQV0uAi-wsvEQ"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 1,
                  "publishedAt": "2024-03-18T16:07:50Z",
                  "updatedAt": "2024-03-18T16:07:50Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "iv-TWMm-IDWGHipLjOi_xGF7WDg",
         "id": "Ugzkn1BbkdlAGLerELd4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "7uuPm69dLYi3kihrfJEr6du9cjw",
               "id": "Ugzkn1BbkdlAGLerELd4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "P R O M O S M",
                  "textOriginal": "P R O M O S M",
                  "authorDisplayName": "@ignazmazanec7639",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/5S3UBy3xQOpYtD8xqu_gezxFZYcwc1KKOHcGBBp3z4hm892tEYSRvPtotw3qthAAAZ_1rWYJ=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@ignazmazanec7639",
                  "authorChannelId": {
                     "value": "UCWz2jcCXzDWtWgvHuvjR_IA"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 0,
                  "publishedAt": "2024-03-15T09:20:43Z",
                  "updatedAt": "2024-03-15T09:20:43Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "HqNtflWxcpZj_25-DcTVWU6VBgQ",
         "id": "UgyybEMMlxu5VJGf41V4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "oufsjJUSgv_EgjqLPCLgStvNJrA",
               "id": "UgyybEMMlxu5VJGf41V4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "고요비 존나 이뻐",
                  "textOriginal": "고요비 존나 이뻐",
                  "authorDisplayName": "@user-gs3hj4up4u",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/o9DmygAlLaWbxJhqEOrX1fuqRHKHXFILwGZY-tW-4MaC3p_LsnZD2hr5cX70JEpQOgyxFTEB=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@user-gs3hj4up4u",
                  "authorChannelId": {
                     "value": "UCbTacrCG97seiAbvczfDEbA"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 1,
                  "publishedAt": "2024-03-10T11:01:06Z",
                  "updatedAt": "2024-03-10T11:01:06Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "rkztsopZvkiJeeg20NZ6PlnaZWI",
         "id": "Ugz7FI-DW7DMruAWUl94AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "ypRiTMXS6TFHj2BiMglr8JMKP14",
               "id": "Ugz7FI-DW7DMruAWUl94AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "영상언제마다올라오나요요ㅠㅠㅠㅠㅠㅠㅠㅠ사실 매일 기다리구있어요ㅠㅠ😢❤",
                  "textOriginal": "영상언제마다올라오나요요ㅠㅠㅠㅠㅠㅠㅠㅠ사실 매일 기다리구있어요ㅠㅠ😢❤",
                  "authorDisplayName": "@Dodod01",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/n8aKgLJznFJhXJICDtB0dBiv9HPOWNr73-GPOsr4aZL_ndN0NCfEJ5ZhLjhlSjbuq4nhNof72Xk=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@Dodod01",
                  "authorChannelId": {
                     "value": "UCnXCivMHIfpDexaO_T_ECBw"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 0,
                  "publishedAt": "2024-02-26T08:14:03Z",
                  "updatedAt": "2024-02-26T08:14:03Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "-Q3Myb3en68Tfm--VSv9MVlB7ro",
         "id": "Ugy-AcwzsU4eeeQR6Cd4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "92EdI394l5nyQEZd9dDkA6s7kEE",
               "id": "Ugy-AcwzsU4eeeQR6Cd4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "고요비..그저 빛<br>고요비 ㅈㄴ멋짐<br>고요비형 그래서 다음영상은 언제?<br>돌아와줘 나만의 바리스타<br>마지막으로 한마디 하자면<br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br>고요비 실물 개이쁨 당장 달려가셈",
                  "textOriginal": "고요비..그저 빛\n고요비 ㅈㄴ멋짐\n고요비형 그래서 다음영상은 언제?\n돌아와줘 나만의 바리스타\n마지막으로 한마디 하자면\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n고요비 실물 개이쁨 당장 달려가셈",
                  "authorDisplayName": "@user-lf8et7yf1d",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/ytc/AIdro_khLtRTQEC3s-G1RWI5cKtOl6Xry7OtP-hBX8HuMdTcGA=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@user-lf8et7yf1d",
                  "authorChannelId": {
                     "value": "UCXSWDoneoy7lWbwCURWJlkg"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 0,
                  "publishedAt": "2024-02-25T18:31:23Z",
                  "updatedAt": "2024-02-25T18:31:23Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "UBeJrlEv00Y9GmQvE0yqZNtebZg",
         "id": "UgyLJHKYAarFVHhjh454AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "WtQNQESWPWb-rfBv5LGFBdp5ieQ",
               "id": "UgyLJHKYAarFVHhjh454AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "요비쌤 빨리 영상좀….미칠꺼같아요옥,,,요비쌤 보고싶어요,,",
                  "textOriginal": "요비쌤 빨리 영상좀….미칠꺼같아요옥,,,요비쌤 보고싶어요,,",
                  "authorDisplayName": "@user-kq8gc6of4l",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/ytc/AIdro_kXziv4EOZpLIyA1LXgcw_RNWK8LIAsIxR4suJpskxV9w=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@user-kq8gc6of4l",
                  "authorChannelId": {
                     "value": "UCcW_kyxtRndSBv-WeTDHVNA"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 1,
                  "publishedAt": "2024-02-24T15:20:24Z",
                  "updatedAt": "2024-02-24T15:20:24Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "LsrTW53wNAP5dni7iT-3JKmXBV0",
         "id": "Ugy7InJuCKI4lMeg6MB4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "muqhFeKp6wrcPExAalg12eCB2xo",
               "id": "Ugy7InJuCKI4lMeg6MB4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "카페 이름하고 위치 알 수 있을까요?",
                  "textOriginal": "카페 이름하고 위치 알 수 있을까요?",
                  "authorDisplayName": "@user-xr3dz8eq8w",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/NamPDFcJYwZh2FFOc9geEMyNViF4WMrx3QtbmmO36cVDOtHp_yMAW6dP5D58rCruK6PKR-ejOqk=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@user-xr3dz8eq8w",
                  "authorChannelId": {
                     "value": "UC-KmHRmCvm0BIm54vfq22-Q"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 0,
                  "publishedAt": "2024-02-24T07:01:40Z",
                  "updatedAt": "2024-02-24T07:01:40Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "X3VmcIvJ1-ytMJihKUFIO_GpYuw",
         "id": "UgyMgx33i5M3yxstLqd4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "fao6SQWdVfcwdsnqlKhmZykDmyo",
               "id": "UgyMgx33i5M3yxstLqd4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "고요비 여행브이로그 존버단 여기요~!! 일상브이로그 좋아요오ㅎㅎ",
                  "textOriginal": "고요비 여행브이로그 존버단 여기요~!! 일상브이로그 좋아요오ㅎㅎ",
                  "authorDisplayName": "@mumble_be9453",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/g0cuaT3wEoUYV5lhnRGAKb0Oujox6-yYfPFsBbr8OwGJsksRvS0SS1p48806UOkoJ1-BEmH_-w=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@mumble_be9453",
                  "authorChannelId": {
                     "value": "UCmy0UxC0wjYacinvZdEsjFA"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 0,
                  "publishedAt": "2024-02-23T10:19:02Z",
                  "updatedAt": "2024-02-23T10:19:02Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "ilgSrjWyD10snZsBOkV2Jd37XaY",
         "id": "Ugy2cvoN3nuZfJKV_9p4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "KxpP8CKhNBxqqPHOfcxoPFTLa5U",
               "id": "Ugy2cvoN3nuZfJKV_9p4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "혹시 영상은 언제언제올라오나요 저.. 매일매일 목빠지게 숨참구기다려요ㅠㅠㅠㅠㅠ😢❤ 사랑해고요비₊˚⊹ᡣ𐭩",
                  "textOriginal": "혹시 영상은 언제언제올라오나요 저.. 매일매일 목빠지게 숨참구기다려요ㅠㅠㅠㅠㅠ😢❤ 사랑해고요비₊˚⊹ᡣ𐭩",
                  "authorDisplayName": "@Dodod01",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/n8aKgLJznFJhXJICDtB0dBiv9HPOWNr73-GPOsr4aZL_ndN0NCfEJ5ZhLjhlSjbuq4nhNof72Xk=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@Dodod01",
                  "authorChannelId": {
                     "value": "UCnXCivMHIfpDexaO_T_ECBw"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 0,
                  "publishedAt": "2024-02-23T08:36:46Z",
                  "updatedAt": "2024-02-23T08:36:46Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 1,
            "isPublic": true
         }
      },
      {
         "kind": "youtube#commentThread",
         "etag": "4dsmobu9GUw_4gKAtEWVBeUtlXs",
         "id": "UgxBcxPfJs4c_OXA-ul4AaABAg",
         "snippet": {
            "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
            "videoId": "-zXoBeDvLS0",
            "topLevelComment": {
               "kind": "youtube#comment",
               "etag": "CxZpvTzZmUiOyaLiEv8lUdjiMfM",
               "id": "UgxBcxPfJs4c_OXA-ul4AaABAg",
               "snippet": {
                  "channelId": "UCASXEOh4LJxWdk2hZnsfNYw",
                  "videoId": "-zXoBeDvLS0",
                  "textDisplay": "얼그레이케이크맛잇어용😂",
                  "textOriginal": "얼그레이케이크맛잇어용😂",
                  "authorDisplayName": "@user-dn6ze8dl7y",
                  "authorProfileImageUrl": "https://yt3.ggpht.com/ytc/AIdro_lYn1S6CnD0uf_l6YRkkhZ6DGh2PBRV61FUYA=s48-c-k-c0x00ffffff-no-rj",
                  "authorChannelUrl": "http://www.youtube.com/@user-dn6ze8dl7y",
                  "authorChannelId": {
                     "value": "UCikr_RXJ7W0oT7TjWq1TzOA"
                  },
                  "canRate": true,
                  "viewerRating": "none",
                  "likeCount": 0,
                  "publishedAt": "2024-02-21T03:12:55Z",
                  "updatedAt": "2024-02-21T03:12:55Z"
               }
            },
            "canReply": true,
            "totalReplyCount": 0,
            "isPublic": true
         }
      }
   ]
}
