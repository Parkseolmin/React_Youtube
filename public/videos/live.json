{
  "kind": "youtube#searchListResponse",
  "etag": "0PqwbvWL-Ah6F17khpm_VzNHxbQ",
  "regionCode": "KR",
  "pageInfo": {
    "totalResults": 7,
    "resultsPerPage": 7
  },
  "items": [
    {
      "kind": "youtube#searchResult",
      "etag": "NFTSM_d3D-E2IZ6h1TtAub61blo",
      "id": {
        "kind": "youtube#video",
        "videoId": "KBpqlmNl8iU"
      },
      "snippet": {
        "publishedAt": "2023-12-01T22:34:03Z",
        "channelId": "UC7e0ym-mN3Hv2UtJalCxj-g",
        "title": "실시간 지진관측",
        "description": "Track provided by Lofi Girl Listen: bit.ly/lofigirI-playlists https://yt.nyj36.xyz.",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/KBpqlmNl8iU/default_live.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/KBpqlmNl8iU/mqdefault_live.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/KBpqlmNl8iU/hqdefault_live.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "NYJ36",
        "liveBroadcastContent": "live",
        "publishTime": "2023-12-01T22:34:03Z"
      }
    },
    {
      "kind": "youtube#searchResult",
      "etag": "gS4Xk7-lUdX4U5jXxRmk6ABNNVQ",
      "id": {
        "kind": "youtube#video",
        "videoId": "ySzcaRmGLo4"
      },
      "snippet": {
        "publishedAt": "2024-05-31T07:38:30Z",
        "channelId": "UC8ICadaEiWPfoqNk1T0GCrA",
        "title": "𝐏𝐥𝐚𝐲𝐥𝐢𝐬𝐭 아침에 듣기 좋은 상쾌한 팝송 | 이거 들으면 무조건 바람 쐬러 가고 싶을걸",
        "description": "Playlist 아침에 듣기 좋은 상쾌한 팝송 | 이거 들으면 무조건 바람 쐬러 가고 싶을걸 #플레이리스트 #playlist #pop #음악 ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/ySzcaRmGLo4/default_live.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/ySzcaRmGLo4/mqdefault_live.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/ySzcaRmGLo4/hqdefault_live.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "항상 기분 | mood",
        "liveBroadcastContent": "live",
        "publishTime": "2024-05-31T07:38:30Z"
      }
    },
    {
      "kind": "youtube#searchResult",
      "etag": "-yhVdaqUfexwpGWckzapO4n2LI4",
      "id": {
        "kind": "youtube#video",
        "videoId": "XmYBF4_UhEg"
      },
      "snippet": {
        "publishedAt": "2024-06-01T08:01:41Z",
        "channelId": "UC8G3q6X-aL-pCNgX18MrqHw",
        "title": "지금라이프(REXTON 더블랙) 서울시 강남구 20240602Sunday 렉스턴 뉴 아레나! 실키드라이빙.♡",
        "description": "렉스턴을 타고 최고의 정숙성을 느껴본다!",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/XmYBF4_UhEg/default_live.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/XmYBF4_UhEg/mqdefault_live.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/XmYBF4_UhEg/hqdefault_live.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "지금라이프(REXTON)AUTONOW",
        "liveBroadcastContent": "live",
        "publishTime": "2024-06-01T08:01:41Z"
      }
    },
    {
      "kind": "youtube#searchResult",
      "etag": "zeYpDZEiMkJndcrNbAMNLPftHkE",
      "id": {
        "kind": "youtube#video",
        "videoId": "HPEKoCpc0vM"
      },
      "snippet": {
        "publishedAt": "2024-06-02T08:46:43Z",
        "channelId": "UC3UycFY7SIGNrlpCxvPAptA",
        "title": "20240602 레퀴엠 2024 공연 (대관)",
        "description": "",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/HPEKoCpc0vM/default_live.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/HPEKoCpc0vM/mqdefault_live.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/HPEKoCpc0vM/hqdefault_live.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "GEEK티비 [긱:TV]",
        "liveBroadcastContent": "live",
        "publishTime": "2024-06-02T08:46:43Z"
      }
    },
    {
      "kind": "youtube#searchResult",
      "etag": "RRtBM5IzCym2w5rhMvbTivzAK2w",
      "id": {
        "kind": "youtube#video",
        "videoId": "1RZh21P43vU"
      },
      "snippet": {
        "publishedAt": "2024-06-02T08:16:47Z",
        "channelId": "UCTPmot8pw4OxnSYbwSfBh8w",
        "title": "Seoul Korea Live Csm",
        "description": "",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/1RZh21P43vU/default_live.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/1RZh21P43vU/mqdefault_live.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/1RZh21P43vU/hqdefault_live.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "지오스타(GEOSTAR)",
        "liveBroadcastContent": "live",
        "publishTime": "2024-06-02T08:16:47Z"
      }
    },
    {
      "kind": "youtube#searchResult",
      "etag": "g0FbTxjSaK53usfJQC09ev5ClrU",
      "id": {
        "kind": "youtube#video",
        "videoId": "YInDT2qSuqg"
      },
      "snippet": {
        "publishedAt": "2024-04-12T03:42:48Z",
        "channelId": "UC8ICadaEiWPfoqNk1T0GCrA",
        "title": "𝐏𝐥𝐚𝐲𝐥𝐢𝐬𝐭 마음을 상쾌하게 해줄 청량한 팝송 플레이리스트",
        "description": "신선한 팝송을 찾고 계십니까? 기분을 바꿔주는 신선한 팝 플레이리스트! 안녕하세요, 여러분! 저희 음악 채널에 방문해주셔서 감사 ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/YInDT2qSuqg/default_live.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/YInDT2qSuqg/mqdefault_live.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/YInDT2qSuqg/hqdefault_live.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "항상 기분 | mood",
        "liveBroadcastContent": "live",
        "publishTime": "2024-04-12T03:42:48Z"
      }
    },
    {
      "kind": "youtube#searchResult",
      "etag": "cEQMwGbEBL-OrLGshhHdV9muj5g",
      "id": {
        "kind": "youtube#video",
        "videoId": "4SzjHQGGc5w"
      },
      "snippet": {
        "publishedAt": "2024-05-22T03:26:15Z",
        "channelId": "UC8ICadaEiWPfoqNk1T0GCrA",
        "title": "𝐏𝐥𝐚𝐲𝐥𝐢𝐬𝐭 평온한 날들을 위한 감성적인 올드 팝송 리스트 | 역대 최고의 팝 음악",
        "description": "플레이리스트 #Playlist #pop #플레이리스트 #music #cafemusic #song #음악 #summer Playlist 평온한 날들을 위한 ...",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/4SzjHQGGc5w/default_live.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/4SzjHQGGc5w/mqdefault_live.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/4SzjHQGGc5w/hqdefault_live.jpg",
            "width": 480,
            "height": 360
          }
        },
        "channelTitle": "항상 기분 | mood",
        "liveBroadcastContent": "live",
        "publishTime": "2024-05-22T03:26:15Z"
      }
    }
  ]
}
