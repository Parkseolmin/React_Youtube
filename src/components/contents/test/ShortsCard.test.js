import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes, useLocation } from 'react-router-dom';
import { default as ShortsCard } from '../ShortsCard';
import { fakevideo as video } from 'test/videos';

describe('ShortsCard Test', () => {
  it('renders grid type correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ShortsCard video={video} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders list type correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ShortsCard video={video} type='list' />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('navigates to detailed video page with video state when clicked', () => {
    function LocationStateDisplay() {
      return <pre>{JSON.stringify(useLocation().state)}</pre>;
    }

    render(
      <MemoryRouter>
        <Routes>
          <Route path='/' element={<ShortsCard video={video} />} />
          <Route
            path={`/video/${video.id}`}
            element={<LocationStateDisplay />}
          />
        </Routes>
      </MemoryRouter>
    );

    const card = screen.getByRole('listitem');
    fireEvent.click(card);

    expect(screen.getByText(JSON.stringify({ video }))).toBeInTheDocument();
  });
});
