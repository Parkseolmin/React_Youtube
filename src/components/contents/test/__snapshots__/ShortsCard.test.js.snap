// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortsCard Test renders grid type correctly 1`] = `
<DocumentFragment>
  <li
    class=""
  >
    <img
      alt="title"
      class="w-full play__icon rounded-lg"
      src="https://www.youtube.com/"
    />
    <div>
      <p
        class="text-sm VideoCard__title font-semibold my-2 line-clamp-2 text-slate-200"
      >
        title
      </p>
      <p
        class="text-xs  my-0"
      >
        channel title
      </p>
      <p
        class="text-xs my-0"
      >
        just now
      </p>
    </div>
  </li>
</DocumentFragment>
`;

exports[`ShortsCard Test renders list type correctly 1`] = `
<DocumentFragment>
  <li
    class="flex gap-1 mx-2 mb-2  relativeVideo"
  >
    <img
      alt="title"
      class="w-60 mr-2 play__icon rounded-md"
      src="https://www.youtube.com/"
    />
    <div>
      <p
        class="text-sm VideoCard__title font-semibold my-2 line-clamp-2 text-slate-200"
      >
        title
      </p>
      <p
        class="text-xs  my-0"
      >
        channel title
      </p>
      <p
        class="text-xs my-0"
      >
        just now
      </p>
    </div>
  </li>
</DocumentFragment>
`;
